lib/poetry.js

```diff
@@ -21,12 +21,30 @@ async function pyprojectTomlToRequirements(modulePath, pluginInstance) {
     generateRequirementsProgress = progress.get(
       'python-generate-requirements-toml'
     );
-    generateRequirementsProgress.update(
-      'Generating requirements.txt from "pyproject.toml"'
+  }
+
+  const emitMsg = (msg) => {
+    if (generateRequirementsProgress) {
+      generateRequirementsProgress.update(msg)
+      log.info(msg);
+    } else {
+      serverless.cli.log(msg);
+    }
+  }
```

Comment:
I am less certain about this part. I extracted what looked like the logging logic.